sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Core.TestFramework
+{
+    /// <summary>
+    /// Encapsulates a process running an instance of the Test Proxy as well as providing access to the Test Proxy administration actions
+    /// via the TestProxyRestClient.
+    /// <seealso href="https://github.com/Azure/azure-sdk-tools/tree/main/tools/test-proxy"/>
+    /// </summary>
+    public class TestProxy
+    {
+        private static readonly string s_dotNetExe;
+
+        public const string DevCertIssuer = "CN=localhost";
+
+        public const string IpAddress = "127.0.0.1";
+
+        public int? ProxyPortHttp => _proxyPortHttp;
+
+        public int? ProxyPortHttps => _proxyPortHttps;
+
+        private readonly int? _proxyPortHttp;
+        private readonly int? _proxyPortHttps;
+        private readonly Process _testProxyProcess;
+        internal TestProxyRestClient Client { get; }
+        private readonly StringBuilder _errorBuffer = new();
+
+        static TestProxy()
+        {
+            string installDir = Environment.GetEnvironmentVariable("DOTNET_INSTALL_DIR");
+            var dotNetExeName = "dotnet" + (Path.DirectorySeparatorChar == '/' ? "" : ".exe");
```

Comment:
That's a rather indirect way of checking for Windows. Use the platform check in System.Runtime.CompilerServices or even Environment.OSPlatform, which should work in this case for Windows (the latter doesn't work for linux/max reliably), which is why the former was created).